src/app/containers/SendInvoke/SendInvokeReadonly.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component } from 'react'
+
+import { connect } from 'react-redux'
+
+import Button from 'preact-material-components/Button'
+import 'preact-material-components/Button/style.css'
+import 'preact-material-components/Theme/style.css'
+
+import { callInvoke } from '../../utils/neonWrappers';
+
+import style from './SendInvoke.css'
+import globalStyle from '../../components/ContentWrapper/ContentWrapper.css'
+
+@connect(
+  state => ({
+    network: state.network,
+    account: state.account
+  })
+)
+
+export default class SendInvokeReadonly extends Component {
+  state = {
+    loading: false,
+    errorMsg: '',
+    txid: ''
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault()
+    const {network, account, transaction, onSuccess } = this.props
+
+    this.setState({
+      loading: true,
+      errorMsg: '',
+      txid: ''
+    })
+
+    const config = {
+      scriptHash: transaction.scriptHash,
+      operation: transaction.operation,
+      arg1: transaction.args[0],
+      arg2: transaction.args[1],
+      amount: transaction.amount,
+      assetType: transaction.type
+    }
+
+    callInvoke(network, account, config)
+      .then((c) => {
+        if (c.response.result === true) {
+          this.setState({
+            loading: false,
+            txid: c.response.txid
+          })
+
+          if (onSuccess) {
+            onSuccess(c.response)
+          }
+        } else {
+          this.setState({
+            loading: false,
+            errorMsg: 'Invoke failed'
+          })
+        }
+      })
+      .catch((e) => {
+        console.log('e', e)
+        this.setState({
+          loading: false,
+          errorMsg: 'Invoke failed'
+        })
+      })
+  }
+
+  render() {
+    const { loading, txid, errorMsg } = this.state
+    const { transaction } = this.props
+
+    return (
+      <div>
```

Comment:
I pointed this out in my big issue the other day, but it's a nice way to break up your concerns if you keep your view JSX in a different component than the connected portion which deals with the redux store. This makes testing easier, separates the implementation from redux in case you want to use the component in a non-connected way, want to remove redux and use a different state management system, or want to keep your testing simple.

This would also allow the component to be written as a stateless functional component which is really lightweight and easy to test.

Reply:
Are you saying we should do something like neon-wallet does?
https://github.com/CityOfZion/neon-wallet/blob/dev/app/containers/DisplayWalletKeys/index.js
https://github.com/CityOfZion/neon-wallet/blob/dev/app/containers/DisplayWalletKeys/DisplayWalletKeys.jsx

That separates redux and I'm ok with that. Just haven't planned out how to organize everything quite yet. We could follow their pattern or do something a bit different.

Or you're saying we should completely separate all jsx from even the react state?

Reply:
Yeah, the way neon-wallet does it is fine!

Reply:
@austinhinderer and I have agreed we'll tackle this as part of refactoring when testing is in place.